src/main/java/com/facebook/openwifirrm/optimizers/tpc/MeasurementBasedApClientTPC.java

```diff
@@ -239,13 +239,25 @@ public Map<String, Map<String, Integer>> computeTxPowerMap() {
 				);
 				continue;
 			}
-			// TODO: only using first radio
-			final int radioIndex = 0;
-
-			int newTxPower =
-				computeTxPowerForDevice(serialNumber, state, radioIndex);
 			Map<String, Integer> radioMap = new TreeMap<>();
-			radioMap.put(UCentralConstants.BAND_5G, newTxPower);
+
+			for (JsonObject radio: state.radios) {
+				Integer currentChannel =
+					radio.has("channel") && !radio.get("channel").isJsonNull()
+						? radio.get("channel").getAsInt()
+						: null;
+				if (currentChannel == null) {
+					continue;
+				}
+				String band = UCentralUtils.getBandFromChannel(currentChannel);
```

Comment:
Doesn't this object already have a "band" field you can use directly?

Reply:
No, because this is from the state which only has channel. The device status has band but not bssid.

Reply:
OK - would still be a lot better to read the band from device config - do we know if the array indexes are correlated?

Reply:
They may be, but I don't know how to confirm it. Also, why is it better to read from device config?

Reply:
Why would you indirectly infer the band based on another field when the band is explicitly written?

Reply:
Makes sense.

I guess @wuweicaifb can look in uCentral repo to see how they generate the device status (which they call config I think) vs how they generate the state (which they call stats (I showed you an example, I think called "nat_stats.json" or something)). Maybe you can tell if the elements are the same or correlated in some way.

The caveat is that if they change this later, we could be in trouble... since there is no explicit definition api/yaml as far as I know.

Reply:
Is `DeviceWithStatus` the device config you mentioned? It has no `band` as an attribute. I think it's safe here to infer the band from channel as long as the data is consistent, but I agree if the band is explicitly provided, directly reading is better. I will file a separate task to investigate how we could add band in a decent way and do it in a new PR.

Reply:
Yes, it's in `DeviceWithStatus.configuration.radios[n].band`. Can you add a TODO comment in this code here as well as `getBandFromChannel()` to remove it later?

Reply:
Discussed this with @RockyMandayam2 and we confirmed they are currently NOT correlated. So no need for TODO, this can stay. Sorry for the thrash.